src/dss/purge.py

```diff
@@ -0,0 +1,47 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import Namespace
+
+from dss.config import DSS_NAMESPACE
+from dss.logger import setup_logger
+from dss.utils import does_namespace_exist, wait_for_namespace_to_be_deleted
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def purge(lightkube_client: Client) -> None:
+    """
+    Removes all notebooks and DSS components. This is done by removing the
+    `dss` namespace, and thus all resources living in that namespace.
+
+    Args:
+        lightkube_client (Client): The Kubernetes client.
+    """
+    try:
+        if not does_namespace_exist(lightkube_client, DSS_NAMESPACE):
+            logger.warn(
+                f"Cannot purge DSS components. Namespace `{DSS_NAMESPACE}` does not exist."
+            )
+            logger.info("You might want to run")
+            logger.info("  dss status      to check the current status")
+            logger.info("  dss logs --all  to review all logs")
+            logger.info("  dss initialize  to install dss")
+            return
+        else:
+            lightkube_client.delete(Namespace, DSS_NAMESPACE)
+            # need to wait on namespace deletion to be completed
+            wait_for_namespace_to_be_deleted(lightkube_client, DSS_NAMESPACE)
+            logger.info(
+                "Success: All DSS components and notebooks purged successfully from the Kubernetes cluster."  # noqa E501
+            )
+    except ApiError as err:
```

Comment:
This funtion should be reused in dss initialize when executed with `--purge` I think we should remove this big try and catch and just do try and catch in the main with the error message.

Reply:
+1

Reply:
Pushed a commit for this. Regarding unit testing logic in `main.py` file, I gave it shot but couldn't write tests that work. As I messaged @misohu about it, I think that you can't just run click commands and that's why I was getting some weird errors (https://stackoverflow.com/questions/53203500/unittest-for-click-module). I tried using the referenced module but I 'm not sure how to mock things since mocking with MagicMock() didn't work when command run in that CliRunner. Maybe we should have an issue for adding unit tests for main, and tackle this in a separate effort.

For reference, the [errors](https://pastebin.canonical.com/p/vz853t9Ms4/) I was getting.

Reply:
lets open issue for these missing tests in main

Reply:
https://github.com/canonical/data-science-stack/issues/82